diff --git a/src/invidious/views/user/preferences.ecr b/src/invidious/views/user/preferences.ecr
index dfda1434..4bad6ec4 100644
--- a/src/invidious/views/user/preferences.ecr
+++ b/src/invidious/views/user/preferences.ecr
@@ -39,11 +39,7 @@
 
             <div class="pure-control-group">
                 <label for="speed"><%= translate(locale, "preferences_speed_label") %></label>
-                <select name="speed" id="speed">
-                    <% {2.0, 1.75, 1.5, 1.25, 1.0, 0.75, 0.5, 0.25}.each do |option| %>
-                        <option <% if preferences.speed == option %> selected <% end %>><%= option %></option>
-                    <% end %>
-                </select>
+                <input name="speed" id="speed" step="0.1" min="0" max="5" inputmode="decimal" type="number" value="<%= preferences.speed %>"/>
             </div>
 
             <div class="pure-control-group">
